Description: add object type check for AES key
 Make sure the key string is a string.
Origin: upstream, http://git.ghostscript.com/?p=ghostpdl.git;h=2118711
Author: Chris Liddell <chris.liddell@artifex.com>
Forwarded: yes
Bug: http://bugs.ghostscript.com/show_bug.cgi?id=699802
Last-Update: 2018-10-18
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/psi/zfaes.c
+++ b/psi/zfaes.c
@@ -45,7 +45,7 @@
     check_dict_read(*op);
     if (dict_find_string(op, "Key", &sop) <= 0)
         return_error(gs_error_rangecheck);
-
+    check_type(*sop, t_string);
     s_aes_set_key(&state, sop->value.const_bytes, r_size(sop));
 
     /* extract the padding flag, which defaults to true for compatibility */
